mlos_bench/mlos_bench/optimizers/mlos_core_optimizer.py

```diff
@@ -104,6 +104,9 @@ def bulk_register(self,
         df_scores = self._adjust_signs_df(
             pd.DataFrame([{} if score is None else score for score in scores]))
 
+        # Convert all score columns to numeric, coercing errors to NaN
+        df_scores = df_scores.apply(pd.to_numeric, errors='coerce')
```

Comment:
Was this fixed already in #789 ?

Reply:
should be but I haven't tested

Reply:
@bpkroth yes, this is more heavy-handed version of what we already have in master